src/fileio.h

```diff
@@ -22,41 +24,15 @@ struct LoadingError : public std::exception {
 	std::string message;
 };
 
-/**
- * Base class for reading the contents of a directory.
- * Intended use:
- * - After creation (with the #MakeDirectoryReader function), call #OpenPath to open the directory.
- * - A new entry of the directory is returned with every call to #NextEntry.
- * - At the end, call #ClosePath to allow the object to clean up, and get ready for a next use.
- *
- * Alternatively, use #MakePath to construct a path you want to examine.
- * @ingroup fileio_group
- */
-class DirectoryReader {
-public:
-	DirectoryReader() = default;
-	virtual ~DirectoryReader() = default;
-
-	virtual void OpenPath(const char *path) = 0;
-	virtual const char *NextEntry() = 0;
-	virtual void ClosePath() = 0;
-
-	const char *NextFile();
-	virtual const char *MakePath(const char *directory, const char *fname) = 0;
-
-	virtual bool EntryIsFile() = 0;
-	virtual bool EntryIsDirectory() = 0;
-};
-
-extern const char *DIR_SEP;  ///< Directory separator character.
+constexpr auto DIR_SEP = std::filesystem::path::preferred_separator;  ///< Directory separator character.
```

Comment:
According to the C++ spec, the "character type used by the native encoding of the filesystem: char on POSIX, wchar_t on Windows "
So on Windows this is a different type than on Linux and Mac. You'll need to specify the type as `char` and use some conversion magic.

Reply:
It seems that the DIR_SEP call could be dropped here, since it's already included in the function definition on fileio.cpp

`const std::string &TrackDesignDirectory()
{
	static std::string dir;
	if (dir.empty()) {
		dir = freerct_userdata_prefix();
		dir += DIR_SEP;
		dir += TRACK_DESIGN_DIRECTORY;
		dir += DIR_SEP;
	}
	return dir;
}`

Maybe that could be tried and if not working, revert the change and try to do some conversion magic :)

Reply:
I've tested dropping that and things seem to work just fine. I'll see if it compiles and then fix the merge issues

Reply:
It seems that didn't work... which kind of conversion magic would you suggest? It would also be possible to use plain '/' since it's supported by both Windows and Unix

Reply:
I suppose a simple cast should do it, with a sanity check to make sure the conversion is valid:

```c++
static_assert(std::filesystem::path::preferred_separator > 0 && std::filesystem::path::preferred_separator <= 127, "Filesystem path separator is not a valid ASCII character");
constexpr const char *DIR_SEP = {static_cast<char>(std::filesystem::path::preferred_separator)};  ///< Directory separator character.
```

Or alternatively we could keep a platform-specific define:

```c++
#ifdef _WIN32
constexpr const char *DIR_SEP = "\\";
#else 
constexpr const char *DIR_SEP = "/";
#endif
```

Reply:
> Or alternatively we could keep a platform-specific define

This seems like a worse option in my opinion given that it negates the portability gained by using `std::filesystem`.

Reply:
There's a problem here. It seems that wchar_t and char are not directly convertable unless you use specific functions https://stackoverflow.com/questions/36948253/take-wchar-t-and-put-into-char.

I've set it to '\' since it's supposed to work on both windows and unix machines and await futher feedback